test/phoenix_live_view/integrations/live_view_test.exs

```diff
@@ -602,4 +605,24 @@ defmodule Phoenix.LiveView.LiveViewTest do
                    end
     end
   end
+
+  describe "put_layout" do
+    @tag session: %{live_layout: {LayoutView, "live.html"}}
+    test "from root mount renders within the layout", %{conn: conn} do
+      {:ok, view, html} = live(conn, "/thermo")
+
+      assert html =~
+               ~r|^LAYOUT<article[\S\s]*LIVELAYOUTSTART-1-The temp is: 1[\S\s]*-LIVELAYOUTEND\n</article>$|
+
+      assert render_click(view, :save, "22") =~
+               ~r|^LIVELAYOUTSTART-22-The temp is: 22[\S\s]*-LIVELAYOUTEND\n$|
+
+      GenServer.call(view.pid, {:set, :page_title, "New Title"})
+      assert_receive {_ref, {:title, "New Title"}}
+    end
+
+    test "raises when called outside of mount", %{conn: _conn} do
+      # TODO
```

Comment:
@josevalim we killed the `socket.mounted` field, so would love your thoughts on performing this check without re-introducing the field

Reply:
We killed it because we moved all `mount` specific options to the return tuple. So maybe we should do this instead:

    {:ok, socket, layout: {FooBar, "baz.html"}}

?